docs_2.0/manual_EN/5.deployment-and-installation/2.compile-and-install-nebula-graph/1.install-nebula-graph-by-compiling-the-source-code.md

```diff
@@ -0,0 +1,128 @@
+
+    ```bash
+    $ git clone --branch v2.0.0-alpha https://github.com/vesoft-inc/nebula-graph.git
+    ```
+
+2. Make the `nebula-graph` directory the current working directory.
+
+    ```bash
+    $ cd nebula-graph
+    ```
+
+3. Create a `build` directory and make it the current working directory.
+
+    ```bash
+    $ mkdir build && cd build
+    ```
+
+4. Generate the Makefile with CMake.
+
+    > **NOTE**:
+    >
+    >* The installation path is /user/local/nebula by default. To customize it, add the `-DCMAKE_INSTALL_PREFIX=/your/install/path/` CMake variable in the following command.
+    >* For the instructions of more CMake variables, see [CMake variables](#cmake-variables).
+
+    ```bash
+    $ cmake -DENABLE_BUILD_STORAGE=on -DENABLE_TESTING=OFF -DCMAKE_BUILD_TYPE=Release ..
+    ```
+
+5. Compile Nebula Graph.
+
+    Assuming *cores* is the number of CPU cores and *mem_gb* is the memory size (in GB), to appropriately speed up the compiling, you can use the value of the smaller one between cores and mem_gb/2 as the value of N in the following command.
+
+    ```bash
+    $ make -j{N}
+    ```
+
+6. Install Nebula Graph.
+
+    ```bash
+    $ make install-all
+    ```
+
+## CMake variables
+
+Usage of CMake variables:
+
+```bash
+$ cmake -DVariable=<value> ...
+```
+
+The following CMake variables can be used at the configure (cmake) stage to adjust the compiling settings.
+
+### ENABLE_BUILD_STORAGE
+
+Starting from the 2.0 pre-release, Nebula Graph supports separated compute and storage. The `ENABLE_BUILD_STORAGE` variable is set to `OFF` by default so that the storage service is not installed together with the graph service.
+
+If you are deploying Nebula Graph on a single host for testing, you can set `ENABLE_BUILD_STORAGE` to `ON` to download and install the storage service automatically.
+
+### CMAKE_INSTALL_PREFIX
+
+`CMAKE_INSTALL_PREFIX` specifies the path where the service modules, scripts, configuration files are installed. The default path is `/usr/local/nebula`.
+
+### ENABLE_WERROR
+
+`ENABLE_WERROR` is `ON` by default and it makes all warnings into errors. You can set it to `OFF` if needed.
+
+### ENABLE_TESTING
+
+`ENABLE_TESTING` is `ON` by default and unit tests are built with the Nebula Graph services. If you just need the service modules, set it to `OFF`.
+
+### ENABLE_ASAN
+
+`ENABLE_ASAN` is `OFF` by default and the building of ASan (AddressSanitizer), a memory error detector, is disabled. To enable it, set `ENABLE_ASAN` to `ON`. This variable is intended for Nebula Graph developers.
+
+### CMAKE_BUILD_TYPE
+
+Nebula Graph supports the following building types:
+
+* `Debug`, the default value of `CMAKE_BUILD_TYPE`, indicates building Nebula Graph with the debug info but not the optimization options.
```

Comment:
the debug info  == > the debug information. 

informal words are not recommended.